src/highrl/utils/teacher_checker.py

```diff
@@ -199,23 +224,33 @@ def compute_difficulty(
     Returns:
         Tuple[float, int]: env area difficulty, env obstacles difficulty
     """
+    # Convert robot/goal position to integer for correct operations
     rob_pos = robot.get_position().to_int()
     goal_pos = robot.get_goal_position().to_int()
+
+    # Run BFS to find a path from goal to robot
     is_valid, path = get_path_bfs(
         obstacles,
         width,
         rob_pos,
         goal_pos,
     )
+
+    # If there is no valid path, return infinite difficulty
     if not is_valid:
         return INF, max(0, len(obstacles.obstacles_list) - 4)
 
+    # Sample points on the line between the robot and the goal
     robot_to_goal_points = sample_line_points(rob_pos, goal_pos, step_size=1)
 
     # Add points sampled from the line between the goal and robot to the path
     path.extend(robot_to_goal_points)
 
+    # Get the points representing the convex polygon
     convex_hull_points = convex_hull_compute(path)
+
+    # Compute the area of the convex polygon
     difficulty_area = get_area_of_convex_polygen(convex_hull_points)
 
+    # Note that "2" here is just a constant, its value would not matter.
```

Comment:
If it does not matter then why is it there?
What is it for?

Reply:
Just for backward compatibility and code integrity. If changed, it would cost a very expensive refactor.